src/main/java/com/datapipe/jenkins/vault/buildstep/VaultStepExecution.java

```diff
@@ -0,0 +1,107 @@
+package com.datapipe.jenkins.vault.buildstep;
+
+import java.io.IOException;
+
+import javax.annotation.Nonnull;
+
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.datapipe.jenkins.vault.VaultAccessor;
+import com.datapipe.jenkins.vault.configuration.GlobalVaultConfiguration;
+import com.datapipe.jenkins.vault.configuration.VaultConfiguration;
+import com.datapipe.jenkins.vault.credentials.VaultCredential;
+import com.datapipe.jenkins.vault.exception.VaultPluginException;
+
+import hudson.EnvVars;
+import hudson.Util;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+
+public class VaultStepExecution extends StepExecution {
+    private final transient VaultReadStep readStep;
+
+    private final EnvVars envVars;
+    private final transient Run run;
+    private final transient TaskListener listener;
+
+    VaultStepExecution(@Nonnull final StepContext context, final VaultReadStep readStep) throws IOException, InterruptedException {
+        super(context);
+        this.readStep = readStep;
+        this.run = getRunFromContext();
+        this.listener = getTaskListenerFromContext();
+        this.envVars = run.getEnvironment(listener);
+    }
+
+    @Override
+    public boolean start() {
+        try {
+
+            final String path = Util.replaceMacro(readStep.getPath(), envVars);
+            final int engineVersion = readStep.getEngineVersion();
+            final VaultAccessor vaultAccessor = initVaultAccessor();
+
+            final String value = vaultAccessor.read(path, engineVersion)
+                .getData()
+                .get(Util.replaceMacro(readStep.getKey(), envVars));
+            getContext().onSuccess(value);
+        } catch (VaultPluginException ex) {
+            getContext().onFailure(ex);
+        }
+        return true;
+    }
+
+    @Override
+    public void stop(@Nonnull final Throwable cause) {
+        // no action required on stopping
+    }
+
+    private VaultAccessor initVaultAccessor() {
+        GlobalVaultConfiguration globalVaultConfiguration = GlobalVaultConfiguration.all().get(GlobalVaultConfiguration.class);
```

Comment:
I'd suggest adding a `get` method which simplify the logic
```suggestion
        GlobalVaultConfiguration globalVaultConfiguration = GlobalVaultConfiguration.get();
```

This is how I wrote it for Jira: https://github.com/jenkinsci/jira-plugin/blob/29bc41e588a07253268e7ee3ca46a928b08fbef0/src/main/java/hudson/plugins/jira/JiraGlobalConfiguration.java#L16 but you can definitely use `.all()`